File: public/examples/integrations/toolkits/github/list_review_comments_on_pull_request_example_llm_oai.py
import os
from openai import OpenAI

USER_ID = "you@example.com"
PROMPT = "List review comments on pull request #1 in the ArcadeAI/hello-world repository."
TOOL_NAME = "Github.ListReviewCommentsOnPullRequest"

client = OpenAI(
    base_url="https://api.arcade-ai.com",
    api_key=os.environ.get("ARCADE_API_KEY"))

response = client.chat.completions.create(
    messages=[
        {"role": "user", "content": PROMPT},
    ],
    model="gpt-4o-mini",
    user=USER_ID,
    tools=[TOOL_NAME],
    tool_choice="generate",
)
print(response.choices[0].message.content)

File: public/examples/integrations/toolkits/github/list_review_comments_on_pull_request_example_llm_oai.js
import OpenAI from 'openai';

const USER_ID = 'you@example.com';
const PROMPT = "List review comments on pull request #2 in the ArcadeAI/example repository.";
const TOOL_NAME = 'Github.ListReviewCommentsOnPullRequest';

const client = new OpenAI({
  baseURL: 'https://api.arcade.dev',
  apiKey: process.env.ARCADE_API_KEY
});

const response = await client.chat.completions.create({
  messages: [
    { role: 'user', content: PROMPT }
  ],
  model: 'gpt-4o-mini',
  user: USER_ID,
  tools: [TOOL_NAME],
  tool_choice: 'generate'
});

console.log(response.choices[0].message.content); 